src/compiler/types.ts

```diff
@@ -5665,8 +5665,28 @@ export enum TypeReferenceSerializationKind {
     ObjectType,
 }
 
+
+/** @internal */
+export interface CoreEmitResolver {

```

Comment:
If both `CoreEmitResolver` and `EmitResolver` are internal, why do we need two interfaces? Is it just to allocate a special `EmitResolver` that doesn't implement the whole interface?

Reply:
Yes exactly. From `transformDeclaration` we don't provide the whole resolver, only a small subset that can be reasonably implemented without the full type checker.